src/core/nearfieldreader.cpp

```diff
@@ -0,0 +1,144 @@
+/***************************************************************************
+ nearfieldreader.cpp - NearFieldReader
+
+ ---------------------
+ begin                : 27.08.2023
+ copyright            : (C) 2023 by Mathieu Pellerin
+ email                : mathieu (at) opengis.ch
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "nearfieldreader.h"
+
+#include <QDebug>
+#include <QNdefMessage>
+#include <QNdefNfcTextRecord>
+#include <QNearFieldTarget>
+#include <QUrl>
+#include <qgsmessagelog.h>
+
+NearFieldReader::NearFieldReader( QObject *parent )
+  : QObject( parent )
+{
+  mNearFieldManager = std::make_unique<QNearFieldManager>( this );
+  connect( mNearFieldManager.get(), &QNearFieldManager::targetDetected, this, &NearFieldReader::handleTargetDetected );
+  connect( mNearFieldManager.get(), &QNearFieldManager::targetDetected, this, &NearFieldReader::handleTargetLost );
+}
+
+NearFieldReader::~NearFieldReader()
+{
+#if QT_VERSION < QT_VERSION_CHECK( 6, 0, 0 )
+  mNearFieldManager->setTargetAccessModes( QNearFieldManager::NoTargetAccess );
+#endif
+  mNearFieldManager->stopTargetDetection();
+}
+
+void NearFieldReader::handleTargetDetected( QNearFieldTarget *target )
+{
+  connect( target, &QNearFieldTarget::ndefMessageRead, this, &NearFieldReader::handleNdefMessageRead );
+  connect( target, &QNearFieldTarget::error, this, &NearFieldReader::handleTargetError );
+
+  emit targetDetected( QString( target->uid() ) );
+
+  if ( target->hasNdefMessage() )
+  {
+    mReadString.clear();
+    QNearFieldTarget::RequestId request = target->readNdefMessages();
+    if ( !request.isValid() )
+    {
+      handleTargetError( QNearFieldTarget::NdefReadError, request );
+    }
+  }
+}
+
+void NearFieldReader::handleTargetLost( QNearFieldTarget *target )
+{
+  disconnect( target );
+}
+
+void NearFieldReader::handleNdefMessageRead( const QNdefMessage &message )
+{
+  qInfo() << QStringLiteral( "Received %1 record message(s) from near-field target" ).arg( message.size() );
```

Comment:
Kill before merge I guess

Reply:
@m-kuhn , I have deliberately left this there for now as I think it'd be useful to get as much details as possible from users in the first public version to launch with NFC support. That'll help us better understand any corner scenario we face down the line. I'd be keen to leave it was is for now (we have such qInfo() in other parts of the code).